trio/tests/test_exports.py

```diff
@@ -114,6 +143,169 @@ def no_underscores(symbols):
         assert False
 
 
+# this could be sped up by only invoking mypy once per module, or even once for all
+# modules, instead of once per class.
+@slow
+# see commend on test_static_tool_sees_all_symbols
+@pytest.mark.redistributors_should_skip
+# pylint/jedi often have trouble with alpha releases, where Python's internals
+# are in flux, grammar may not have settled down, etc.
+@pytest.mark.skipif(
+    sys.version_info.releaselevel == "alpha",
+    reason="skip static introspection tools on Python dev/alpha releases",
+)
+@pytest.mark.parametrize("module_name", PUBLIC_MODULE_NAMES)
+@pytest.mark.parametrize("tool", ["jedi", "mypy"])
+def test_static_tool_sees_class_members(tool, module_name, tmpdir) -> None:
+    module = PUBLIC_MODULES[PUBLIC_MODULE_NAMES.index(module_name)]
+
+    # ignore hidden, but not dunder, symbols
+    def no_hidden(symbols):
+        return {
+            symbol
+            for symbol in symbols
+            if (not symbol.startswith("_")) or symbol.startswith("__")
+        }
+
+    if tool == "mypy":
+        if sys.implementation.name != "cpython":
+            pytest.skip("mypy not installed in tests on pypy")
+        # create py.typed file
+        (Path(trio.__file__).parent / "py.typed").write_text("")
```

Comment:
I'm not *thrilled* about this, given... could `trio.__file__` point to a user's development checkout of trio? If there's a very simple alternative I would prefer that, though this is very much a transient issue and easy to work around on a user's side.

Reply:
I can clean it up afterwards if it didn't exist before running